format/format.go

```diff
@@ -245,8 +245,10 @@ func (w *writer) writeRule(rule *ast.Rule, isElse bool, comments []*ast.Comment)
 
 	if len(rule.Head.Args) > 0 {
 		close = closingLoc('(', ')', '{', '}', rule.Location)
-	} else {
+	} else if rule.Head.Key != nil {
```

Comment:
Are we testing this change anywhere? I couldn't see how this was related.

Reply:
Thats for rules like `p[x]` for the head where it needs to skip over a block of `[` to `]` before looking for the `}`. It was _always_ giving back a bad value for `close` with the old code for complete rules... now its only wrong for cases with arrays, sets, or objects in the head keys/values 😢

Its not a huge deal if its wrong, it just messes with the `insertComments` logic in where it decides to place things.

Long story short, it is affecting that end-result in the `test.rego.formatted`. It's part of what makes comments stick to the closing braces now (combined with having the correct text for it to search through)

Reply:
I ended up reverting that change after thinking about it more.. My own comment 

> Its not a huge deal if its wrong, it just messes with the insertComments logic in where it decides to place things.

Convinces me that leaving thing as-is (even if arguably not correct) gives a better end result so we'll just roll with it until we can actually get end locations correctly in here.